### PR TITLE
Update g28 scara

### PR DESCRIPTION
**Pull request by annhan**
- Modify `G28` for SCARA
- Set Home ANGLE
- Add `endstops.checkEndStop` method
- Make `planner.buffer_segment` public

### CODE REVIEW

File: Marlin/Configuration.h [hunk 1]
```diff
@@ -61,12 +61,33 @@
...
+
+    #define THETA_HOMING_OFFSET  0  // Calculated from Calibration Guide and M360 / M114. See http://reprap.harleystudio.co.za/?page_id=1073
+    #define PSI_HOMING_OFFSET    0  // Calculated from Calibration Guide and M364 / M114. See http://reprap.harleystudio.co.za/?page_id=1073
+  #elif ENABLED(MP_SCARA)
+    #define SCARA_OFFSET_THETA1  12 // degrees
+    #define SCARA_OFFSET_THETA2 131 // degrees
+  #endif
+  #define THETA_ANGLE_AT_HOME -30 //THETA DEGREE AT LIMIT SWITCH
+  #define PSI_ANGLE_AT_HOME 135 //PSI DEGREE AT LIMIT SWITCH
+#endif
```
Comment (thinkyhead) [inline Marlin/Configuration.h]: These changes should be submitted to the `MarlinFirmware/Configurations` repo where there's a default SCARA config.

**Comment by thinkyhead** [comment]
SCARA machines vary a lot in terms of how they reckon the home position and deal with endstops. I have a prototype for a homing procedure specific to SCARA that first homes Z (to the top) and then homes XY (AB) together until one stop is hit, and then homes the remaining axis. The current position is then decided based on the AB angles and a forward-kinematics calculation specific to the machine.

**Comment by sl1pkn07** [comment]
tested this PR in my new printer (https://github.com/madl3x/x-scara) (using MORGAN config)

"X" and "Y" home (auto home) ok (G28 halt the printer if not apply this PR), Z not move

individual homig X and Y not work, and individual Z moves X and Y randomly

greetings

**Comment by thinkyhead** [comment]
This will need to be updated to fit the current code-base, and some standardization of SCARA is also needed. Please coordinate with other SCARA enthusiasts to help improve the situation.

**Comment by chepo92** [comment]
SCARA (and all robotics) enthusiast here , @annhan , we can collaborate to get this and mine (https://github.com/MarlinFirmware/Marlin/pull/21005) merged, 
Also we can add documentation so it's more standarized

**Comment by thinkyhead** [comment]
Duplicate of #21285

**Pull request closed without merging**